### ISSUE TITLE
Unable to add member with sub with future deactivation/leave date to new sub

### ISSUE THREAD

**Issue by pascalfree** [juntagrico/juntagrico#394]
Members of deactivated subscriptions or that will have left a subscription (in the future). Should be available to be added to a new subscription (with a join date > leave/deactivation date of old subscription)
 
This is the code in question:
https://github.com/juntagrico/juntagrico/blob/13f7f8797e2191cf52da51451bce2c8224c8047a/juntagrico/admins/inlines/subscription_membership_inlines.py#L48-L58

**Comment by pascalfree** [comment]
While trying to reproduce this, I found another issue:
The `has_subscription` query that is used as well to get `members_for_subscription` (in all versions), is used as "member has a an active subscription" and the negation is expected to be "member has not active subscription at all". But what the negation actually checks is "member has a not active subscription".
https://github.com/juntagrico/juntagrico/blob/3d008e0a37b0b02cea358ee2e4b2f809f049cdc2/juntagrico/dao/memberdao.py#L42-L47
The difference shows in this scenario: A Member has 2 subscription in their record: One in the past and another one that they are currently in. In that case the has_subscription will return the member, but the negation of has_subscription will also return the member (as there is an not active subscription of the member).
I'm not sure how to solve this yet, but it might not be needed at all.
#545 solves a related issue and the conclusion there was, that selectable members can not be prefiltered, as the selectability of the member depends on their join_date and leave_date, which are defined only after their selection and should be evaluated when saving the subscription (as #545 does).

# Conclusion
#545 should be applied to members_for_future_subscription as well. And then the `formfield_for_foreignkey` here would actually collapse, i.e. can be removed. Same for this I think: https://github.com/juntagrico/juntagrico/blob/291e44c00809ade119babf5606cf2a4135bd9d9f/juntagrico/admins/member_admin.py#L33
These are also the only places were the negations of `has_subscription` are used, so the new issue would be resolved. A warning should be added in a comment for the negated use of these method.
